### DISCUSSION TITLE
Querying about supported platforms

### DISCUSSION THREAD

**Discussion by BrawlerXull** [firebase/flutterfire#12525] [Q&A]
I am trying to develop an application using firebase fcm for MacOS
I am running into this error
```
1970-01-02 04:49:38.216263+0300 Runner[675:161670] flutter: [firebase_messaging/apns-token-not-set] APNS token has not been set yet. Please ensure the APNS token is available by calling `getAPNSToken()`.
1970-01-02 04:49:38.216725+0300 Runner[675:161670] flutter: #0      MethodChannelFirebaseMessaging._APNSTokenCheck (package:firebase_messaging_platform_interface/src/method_channel/method_channel_messaging.dart:138:9)
<asynchronous suspension>
#1      MethodChannelFirebaseMessaging.subscribeToTopic (package:firebase_messaging_platform_interface/src/method_channel/method_channel_messaging.dart:390:5)
<asynchronous suspension>
#2      PushNotificationService.subscribeUserToFirebase (package:team/src/common/notification/firebase/push_notification.dart:70:7)
<asynchronous suspension>
#3      userAPI (package:team/src/features/master_data/application/master_data_providers.dart:116:5)
<asynchronous suspension>
#4      FutureHandlerProviderElementMixin.handleFuture.<anonymous closure>.<anonymous closure> (package:riverpod/src/async_notifier/base.dart:337:9)
<asynchronous suspension>
```
I wanted to ask if Firebase FCM supports MacOS and the solution of this Error
I have searched for different solutions on this error and i came to know that I need Apple developer account for this.
I know that FCM supports iOS , but my main concern is does it supports MacOs?
If yes what are steps that I need to perform to set it up for MacOS other than iOS

**Comment by russellwheatley** [comment]
@BrawlerXull - it does support macOS. Check out the firebase [messaging example app](https://github.com/firebase/flutterfire/tree/master/packages/firebase_messaging/firebase_messaging/example) and [follow the documentation](https://firebase.google.com/docs/cloud-messaging/flutter/client?_gl=1*pgwqu4*_up*MQ..*_ga*ODgyNTY1MDE1LjE3MTEzNTU3Njk.*_ga_CW55HF8NVT*MTcxMTM1NTc2OC4xLjAuMTcxMTM1NTc2OC4wLjAuMA..).

**Reply by BrawlerXull** [reply]
> @BrawlerXull - it does support macOS. Check out the firebase [messaging example app](https://github.com/firebase/flutterfire/tree/master/packages/firebase_messaging/firebase_messaging/example) and [follow the documentation](https://firebase.google.com/docs/cloud-messaging/flutter/client?_gl=1*pgwqu4*_up*MQ..*_ga*ODgyNTY1MDE1LjE3MTEzNTU3Njk.*_ga_CW55HF8NVT*MTcxMTM1NTc2OC4xLjAuMTcxMTM1NTc2OC4wLjAuMA..).

Are windows and Linux supported?

**Reply by BrawlerXull** [reply]
hey @russellwheatley as mentioned on the docs page u shared there are docs given only for iOS , Android and Web there is nothing mentioned about MacOS that is why I am confused

<img width="179" alt="Screenshot 2024-03-26 at 2 05 46 AM" src="https://github.com/firebase/flutterfire/assets/111864031/979f8e57-46c3-4378-9121-4e8eede09aca">

**Reply by BrawlerXull** [reply]
@russellwheatley 
please reply

**Reply by russellwheatley** [reply]
If you go to the README.md on FlutterFire, you will see a table of support for each plugin: https://github.com/firebase/flutterfire . Linux is not supported, and windows has some support but is not production ready

**Comment by BrawlerXull** [comment]
@russellwheatley does it supports in MacOS when the App is in background?
I am making a flutter application with Firebase Cloud Messaging and I want it to support MacOS also.
I want to send notification when the application is killed/background state in MacOS will it work?

**Comment by BrawlerXull** [comment]
<img width="1436" alt="Screenshot 2024-04-28 at 1 59 08 AM" src="https://github.com/firebase/flutterfire/assets/111864031/49655a6a-6652-47ee-9402-ca64af86568a">
I am asking it because here it is writtern iOS only

**Comment by BrawlerXull** [comment]
@russellwheatley
